src/timeout.h

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2018-2021 Redis Labs Ltd. and Contributors
+ *
+ * This file is available under the Redis Labs Source Available License Agreement
+ */
+
+#pragma once
+
+#include "execution_plan/execution_plan.h"
+
+typedef struct {
+	ExecutionPlan *plan;    // the query's ExecutionPlan
+	bool query_completed;   // whether the query has finished successfully
+	bool changes_committed; // whether the graph has been locked for commits
+} TimeoutCtx;
```

Comment:
I believe we can remove both `query_completed` and `changes_committed`
Once a write query tries to acquire the write-lock for the first time, it should (before acquiring the lock) call `ClearTimeout` if it succeed in clearing the timeout (given that a timeout is present) the query can proceed, otherwise the timeout been triggered, it shouldn't acquire the lock and return exception.